### ISSUE TITLE
Ask user before installing protocol handler for SEP-7

### ISSUE THREAD

**Issue by andywer** [satoshipay/solar#957]
Nikhil had a good point: He said that the installation of the `web+stellar` protocol handler is somewhat like choosing your default browser. The user should be asked to opt-in.

Now the first big question is: **Is it possible to install the protocol handler during app run time and only after user interaction on all platforms?**

**Comment by andywer** [comment]
@ebma For Electron there is no way to install the protocol handler programatically – you define the protocols you want to handle in a manifest file and then the OS will invoke our executable, passing the link as a command-line argument, right?

**Comment by ebma** [comment]
I think it could be possible if we put [this statement](https://github.com/satoshipay/solar/blob/2fec982ede2ab09402937f5e9947819da1a42706/electron/src/bootstrap.ts#L5) into some condition, e.g. only call this when the user opted in to use the application as the default protocol handler. And if the user opts out again we can just call [app.removeAsDefaultProtocolClient()](https://github.com/satoshipay/solar/blob/2fec982ede2ab09402937f5e9947819da1a42706/electron/src/bootstrap.ts#L5).

And regarding other platforms: IIRC on Android you get to choose between all available applications anyways (see [Figure 1](https://developer.android.com/training/basics/intents/sending#StartActivity)) so I suppose we don't have to do anything here.
I don't know how it works on iOS though.

**Comment by andywer** [comment]
Cool! Then let's call the [setAsDefaultProtocolHandler() statement](https://github.com/satoshipay/solar/blob/2fec982ede2ab09402937f5e9947819da1a42706/electron/src/bootstrap.ts#L5) upon user interaction (opt-in) only for the desktop apps.

For iOS I read somewhere that there is no defined behavior which app it's gonna pick if multiple apps subscribed to the same protocol… 🤦‍♂️

**Comment by andywer** [comment]
Fortunately there is also a method [app.isDefaultProtocolClient()](https://www.electronjs.org/docs/api/app#appisdefaultprotocolclientprotocol-path-args) for Electron, so we know if we should prompt the user or not. And there is [app.getApplicationNameForProtocol(url)](https://www.electronjs.org/docs/api/app#appgetapplicationnameforprotocolurl) to check whether some app handling the protocol is installed already (I think it works like that, didn't test). We then also need to extend the app settings.

- On app launch:
  - Check if we are the default handler
  - Check if there is a default handler installed at all
  - Check if the user prompt was shown before and dismissed
  - If we are not the default handler and no other app is handling Stellar URIs, make us the default handler
  - If we are not the default handler, there is another Stellar URI - handling app and the prompt hasn't been dismissed before, show prompt
- Prompt:
  - Show toast notification "Do you want Solar to handle interactive Stellar links on this computer (recommended)?"
- Add item to app settings (desktop only):
  - If we are the default handler, show as disabled and with a "Solar is handling Stellar links" status
  - If we are not the default handler, show as "make solar handle all Stellar URIs" action, make us the default handler on click

The user prompt is more a nice-to-have. The only-make-us-default-handler-if-there-is-none-yet check and the additional app setting are must-haves.
